python/TestHarness/TestHarness.py

```diff
@@ -311,6 +311,7 @@ def augmentParameters(self, filename, tester):
     #          the leading part of the path
     test_dir = os.path.abspath(os.path.dirname(filename))
     relative_path = test_dir.replace(self.run_tests_dir, '')
+    first_directory = re.split('\W+', relative_path)[1] # Get first directory
```

Comment:
This isn't quite correct. There are many characters in the `\W` set, some of this might match in a valid path giving you extra splits. I suggest you use `os.path.split` or you can get the os portable path separater and split on that inside of a normal regex.